ssz_serialization.nim

```diff
@@ -162,11 +162,36 @@ proc writeVarSizeType(w: var SszWriter, value: auto) {.raises: [Defect, IOError]
     # to internally match the binary representation of SSZ BitLists in memory.
     writeSeq(w, bytes value)
   elif value is object|tuple|array:
-    trs "WRITING OBJECT OR ARRAY"
-    var ctx = beginRecord(w, type value)
-    enumerateSubFields(value, field):
-      writeField w, ctx, astToStr(field), field
-    endRecord w, ctx
+    when isCaseObject(type(value)):
```

Comment:
Not all case objects are valid SSZ unions. You can create a specialized `isUnion(T)` predicate in ssz/types and leave a TODO comment that it should be implemented to match the behavior of the spec.

Case objects that are not SSZ unions can be handled with a compile-time error.

Reply:
Yes, exactly. This is what I meant with the TODO comment here: https://github.com/status-im/nim-ssz-serialization/blob/ssz-union/ssz_serialization/codec.nim#L115

It is my intention to add this still, but I first wanted feedback to see if it is a good way forward to map SSZ Unions on case objects like this.